downsample_rs/src/minmaxlttb/generic.rs

```diff
@@ -21,8 +36,18 @@ where
     // Apply first min max aggregation (if above ratio)
     if x.len() / n_out > minmax_ratio {
         // Get index of min max points
-        let mut index: Array1<usize> =
-            f_minmax(x.slice(s![1..-1]), y.slice(s![1..-1]), n_out * minmax_ratio);
+        // check if function is serial or parallel. if parallel, n_threads cannot be None
+        let mut index = match f_minmax {
+            MinMaxFunctionWithX::Serial(func) => {
+                func(x.slice(s![1..-1]), y.slice(s![1..-1]), n_out * minmax_ratio)
+            }
+            MinMaxFunctionWithX::Parallel(func) => func(
+                x.slice(s![1..-1]),
+                y.slice(s![1..-1]),
+                n_out * minmax_ratio,
+                n_threads.unwrap_or(1),
```

Comment:
It should indeed never be `None`, but I like to add a safeguard just in case. Do we just use the `.unwrap()` then? Crashing would make sense as this would indicate an issue in the source code.

Reply:
I would personally go for `.unwrap()`  and accept a crash as this pinpoints a source-code related issue.